lib/features/mailbox_dashboard/data/model/composer_cache.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:equatable/equatable.dart';
+import 'package:jmap_dart_client/http/converter/email/email_body_value_converter.dart';
+import 'package:jmap_dart_client/http/converter/email/email_mailbox_ids_converter.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_address.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_body_part.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_body_value.dart';
+import 'package:jmap_dart_client/jmap/mail/mailbox/mailbox.dart';
+import 'package:model/email/email_content.dart';
+import 'package:model/extensions/media_type_extension.dart';
+
+class ComposerCache with EquatableMixin {
+
+  final EmailId id;
+  final Map<MailboxId, bool>? mailboxIds;
+  final String? subject;
+  final Set<EmailAddress>? from;
+  final Set<EmailAddress>? to;
+  final Set<EmailAddress>? cc;
+  final Set<EmailAddress>? bcc;
+  final Set<EmailAddress>? replyTo;
+  final Set<EmailBodyPart>? textBody;
+  final Set<EmailBodyPart>? htmlBody;
+  final Map<PartId, EmailBodyValue>? bodyValues;
+
+  ComposerCache(
+      this.id,
+      {
+        this.mailboxIds,
+        this.subject,
+        this.from,
+        this.to,
+        this.cc,
+        this.bcc,
+        this.replyTo,
+        this.textBody,
+        this.htmlBody,
+        this.bodyValues,
+      });
+
+  factory ComposerCache.fromJson(Map<String, dynamic> json) {
```

Comment:
Why don't you use `JsonSerializable` to auto-generate `fromJson` and `toJson`

Reply:
Because it has some unique variables like EmailId, Id, ... this object like Email(in jmap-dart-client) object cant use JsonSerializable

Reply:
Oh. You can create `converter` for `EmailId` to use it.